Remote File Sources and Storage Locations redesign

itisAliRH opened the pull request:
This pull request updates the `Remote File Sources` and `Storage Locations` options lists to enhance user experience through improved navigation and a modernized interface.

Closes #19440

|Before|After|
|--|--|
|![image](https://github.com/user-attachments/assets/a61b0047-4941-47c7-9241-3cafabb8d313)|![image](https://github.com/user-attachments/assets/5e1240d3-1da6-466c-9624-db924f18ad0c)|
|![image](https://github.com/user-attachments/assets/7e99a64b-4a0c-4d4b-97cd-ddd3cdc40402)|![image](https://github.com/user-attachments/assets/62c479d6-846d-4d39-991f-58e37e297f90)|
|![image](https://github.com/user-attachments/assets/24ffa7f6-86d9-4469-bedc-fe3c9bfa1540)|![image](https://github.com/user-attachments/assets/66dbd310-4a11-4294-a7b5-d92bebf809ad)|

Key Changes:
- Replace the current available options list with a cards list view
- Adds navigation breadcrumb across the list, options and new instance pages
- Adds search field to filter options by their name
- The new options list works well on smalleer screen sizes

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:
  1. Go to `User Preferences > Manage Your Storage Locations` OR `User Preferences > Manage Your Remote File Sources`
  2. Click on create button on top right of the page

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

itisAliRH committed:
Move breakpoints SCSS file and  import paths

Moves breakpoints SCSS file to a more appropriate directory for improved project structure and reusability and updates import paths in related components for consistency and clarity

itisAliRH committed:
Adds HTML rendering and expand text options to TextSummary

Introduces `isHtml` prop to render description as HTML
Adds `showExpandText` prop to toggle expand text visibility
Updates styles for expand button and floating behavior

itisAliRH committed:
Refactors ManageIndexHeader component

Replaces BRow and BCol with a flexbox layout
Updates prop type definitions and usage
Removes unused imports and adds Heading component

itisAliRH committed:
Adds FileSourceTemplateCard component

Introduces a new component to display file source templates.
Includes functionality for filtering by type and selecting templates.
Applies responsive design and styling for different screen sizes.

itisAliRH committed:
Enhances file source template selection UI

Adds search and filter functionality for file source templates
Use FileSourceTemplateCard for each available template
Improves error handling and loading indicators
Updates UI components for better user experience

itisAliRH committed:
Adds navigation links and heading to CreateInstance component

Introduces navigation links to 'Remote File Sources' and 'Templates' pages
Adds a heading component for better structure and readability

itisAliRH committed:
Adds file source type details and valid filters

Introduces detailed information for each file source type, including icons and descriptive messages.
Adds valid filters for file sources based on name and type.

itisAliRH committed:
Removes SelectTemplate component

Deletes the SelectTemplate.vue file from the FileSources/Templates directory. This component is no longer needed as part of the refactoring of file source templates.

itisAliRH committed:
Adds user preferences link with icon to CreateInstance component header

itisAliRH committed:
Adds header prop and updates button ID generation in ConfigTemplate header

Introduces a new 'header' prop to ManageIndexHeader component
Updates button ID generation to be dynamic based on the header
Replaces static create button IDs in various components

itisAliRH committed:
Consolidates config template card component

Move FileSourceTemplateCard to the generic SourceTemplateCard component under ConfigTemplates
Unifies template selection handling for file sources and object stores

itisAliRH committed:
Updates template card and icons in CreateUserFileSource

Replaces FileSourceTemplateCard with SourceTemplateCard
Adds user preferences icon and type icon handling
Improves template type label styling and filtering

itisAliRH committed:
Adds heading with navigation links to CreateInstance

Introduces a new heading component with navigation links for better user navigation.
Imports necessary components and updates the template structure.

itisAliRH committed:
Adds object store template types and valid filters

Introduces object store template types with associated icons and messages
Adds valid filters for object store templates based on name and type

itisAliRH committed:
Updates ObjectStoreBadge component to use FontAwesome icons directly

Replaces string-based FontAwesome icon references with direct imports
Refactors badge type interface and prop definitions
Enhances tooltip functionality with dynamic title generation

itisAliRH committed:
Removes unused badge icons import

Deletes the badgeIcons.ts file which contained unused FontAwesome icon imports to clean up the codebase.

itisAliRH committed:
Refactors ObjectStoreBadges component and updates styles

Replaces BadgeType with ObjectStoreBadgeType for better type clarity
Simplifies props definition and removes unused properties
Updates import paths for consistency
Adds scoped styles for improved layout and appearance

itisAliRH committed:
Enhances template selection UI and filtering

Replaces template selection components with new UI elements
Adds filtering functionality for template types
Integrates Bootstrap-Vue and FontAwesome for improved styling
Updates computed properties and methods for filtering logic

itisAliRH committed:
Updates badge type import and usage

Replaces the import and usage of BadgeType with ObjectStoreBadgeType
to improve type clarity and consistency in TemplateSummary.vue.

itisAliRH committed:
Removes unused SelectTemplate component

Deletes the SelectTemplate.vue file as it is no longer needed in the project.

itisAliRH committed:
Removes SelectTemplate component

Deletes the SelectTemplate.vue file from the ConfigTemplates directory.

itisAliRH committed:
Adds headers to object store and file source routes

Includes headers "Storage Locations" and "Remote File Sources"
to the props of the respective routes for better clarity.

itisAliRH committed:
Adds headers to object store and file source routes

Includes headers "Storage Locations" and "Remote File Sources"
to the props of the respective routes for better clarity.

ElectronicBlueberry commented on client/src/style/scss/breakpoints.scss:
The underscore indicates to sass that this file should not be complied to css. Removing it adds an empty css file to the sass output.

itisAliRH committed:
Renames SCSS breakpoints file

Updates SCSS import paths to reference the renamed `_breakpoints.scss` file

ref https://github.com/galaxyproject/galaxy/pull/19521/files#r1954598636

itisAliRH committed:
Adds new remote file sources

itisAliRH committed:
Refactors header to use BreadcrumbHeading component

Replaces Heading component with BreadcrumbHeading for improved UI consistency.
Removes unused imports and adds computed property for breadcrumb items.

itisAliRH committed:
Replaces heading with breadcrumb component

Updates the FileSources/CreateInstance component to use a BreadcrumbHeading component.
Removes unused components.

itisAliRH committed:
Replaces heading with breadcrumb component

Removes unused imports and simplifies template by using BreadcrumbHeading component

itisAliRH committed:
Replaces Heading with BreadcrumbHeading component

Updates navigation structure to use BreadcrumbHeading
Removes unused imports and improves breadcrumb navigation

itisAliRH committed:
Enhances template loading UX in CreateInstance

Wraps loading message in BAlert for better visibility
Localizes loading message for improved user experience

itisAliRH committed:
Renames SourceTemplateCard to SourceOptionCard component

Update  component with improved layout and optional grid view.

itisAliRH committed:
Refactors ManageIndexHeader breadcrumb header structure

Fix the create button ui

itisAliRH committed:
Refactors CreateUserFileSource component to improve option selection and loading feedback

Refactors option selection and loading feedback

Improves user experience by refining option selection process.
Enhances loading feedback and integrates user preferences for view mode.
Simplifies and cleans up the template filtering logic.

itisAliRH committed:
Refactors CreateUserObjectStore component

Simplifies imports and removes unused dependencies
Replaces template terminology with option terminology
Introduces breadcrumb navigation and list view toggle
Streamlines filtering logic and improves readability

itisAliRH committed:
Refactors option type handling

Replaces separate icon and title props with a single optionType prop
for better type safety and code readability.

itisAliRH committed:
Removes redundant SelectTemplate tests

Deletes outdated unit tests for SelectTemplate

itisAliRH committed:
Updates mocks directive bind method

Enhances mocked directive bind method to set data attributes

itisAliRH committed:
Adds badge messages for various storage conditions

Introduces constant messages for different storage states such as restricted access, quota settings, speed, security, stability, and cloud storage. These messages will be used to inform users about the properties and management of their storage locations.

itisAliRH committed:
Refactors ObjectStoreBadge message handling

Moves badge messages to a separate file for better organization
Simplifies template structure by removing redundant span tags

itisAliRH committed:
Migrates ObjectStoreBadge tests to TS and fix tests

Updates test assertions and helper functions accordingly to test tooltip.
Replaces existing JS tests for ObjectStoreBadge with TS tests to improve type safety and maintainability.

itisAliRH committed:
Migrates ObjectStoreBadges tests to TS

Renames ObjectStoreBadges test file to TypeScript
Updates imports to include '.vue' extension
Adds type assertion for shallowMount component
Corrects badge size attribute in test assertions

itisAliRH committed:
Refactors BreadcrumbItem type definition

Moves BreadcrumbItem interface to a shared index file for reuse across components. Simplifies BreadcrumbHeading component by importing the shared type.

itisAliRH committed:
Adds SourceOptionsList component

Introduces SourceOptionsList component to display and filter source options
Implements filtering logic and UI for the list
Includes loading and empty state alerts for better UX
Adds scoped styles for the new component

itisAliRH committed:
Simplifies file source templates component

Replaces multiple individual components with a unified `SourceOptionsList` component.
Moves template initialization to onMounted hook.
Removes unused imports and redundant code.

itisAliRH committed:
Refactors CreateUserObjectStore component

Simplifies component by removing unused imports and filtering logic
Replaces individual elements with SourceOptionsList component for better modularity

itisAliRH committed:
Fixes template typo in SourceOptionsList

itisAliRH committed:
Fixes syntax format error in mockedDirective bind method

itisAliRH committed:
Move badges display logic to SourceOptionsList

Introduces an optional 'showBadges' prop to SourceOptionsList component.
Replaces slot-based badge rendering with ObjectStoreBadges component.
Removes redundant badge rendering from CreateUserObjectStore.

itisAliRH committed:
Updates button selectors in templates

Adds unique IDs to select buttons for better identification in SourceOptionCard
Modifies navigation selectors to use updated button IDs

guerler commented on client/src/components/ConfigTemplates/SourceOptionCard.vue:
```diff
@@ -0,0 +1,139 @@
+@import "theme/blue.scss";
+@import "_breakpoints.scss";
+
+.source-option-card {
```
It seems like we are adding another separate card/grid display including styles here, like previously in panels and grids. We should consolidate these and use a shared component if we can.

itisAliRH replied:
@guerler Thank you for pointing this out. You're right, but this would require a large clean-up (as we have several components using this UI) and code refactoring. I'll create the following PR(s) (before adding new cards and more duplications) to replace all existing cards with generic ones and reduce the redundancy. What do you think?

guerler replied:
We had the same discussion with regard to workflow and notification cards. I don't want to hold this PR up tho. Please follow-up with consolidating those, so cards look consistent and can be updated/fixed without having to change many locations manually.

itisAliRH committed:
Adds Zenodo as a new remote file source

Introduces Zenodo as an option for remote file sources, with an associated icon and descriptive message.

davelopez reviewed (approved):
Looks really good to me! Thank you @itisAliRH!
